### PR TITLE
Bitcoind backend on server start up 

### LINKED ISSUE #1918: Wallet via bitcoind rpc rather than bitcoin-s p2p
Talking with Nicolas Dorier this morning he said it would be _very_ nice to have way to use a DLC wallet with just a normal bitcoind rpc client _and_ a _pruned_ bitcoind rather than p2p neutrino stuff. We should investigate and comment here with what it will take to get this done

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #1918

### CODE REVIEW

File: app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala [hunk 1]
```diff
@@ -29,7 +32,16 @@ case class BitcoindRpcAppConfig(
...
 
-  override def start(): Future[Unit] = FutureUtil.unit
+  lazy val nodeConf: NodeAppConfig = NodeAppConfig(directory, confs: _*)
+
+  override def start(): Future[Unit] = {
+    nodeConf.nodeType match {
+      case NodeType.BitcoindBackend =>
+        client.start().map(_ => ())
+      case NodeType.SpvNode | NodeType.NeutrinoNode | NodeType.FullNode =>
+        FutureUtil.unit
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala:43]: Shouldn't this just be an error?

File: node/src/main/scala/org/bitcoins/node/NodeType.scala [hunk 2]
```diff
@@ -6,21 +6,29 @@ sealed abstract class NodeType {
...
     override def shortName: String = "neutrino"
   }
 
-  final case object SpvNode extends NodeType {
+  final case object SpvNode extends InternalImplementationNodeType {
     override def shortName: String = "spv"
   }
 
-  val all: Vector[NodeType] = Vector(FullNode, NeutrinoNode, SpvNode)
+  final case object BitcoindBackend extends ExternalImplementationNodeType {
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/NodeType.scala:26]: Hmm, i'm not sure if I agree with this being considered a `NodeType`. It's more of a source for chain information. Would you consider the block stream API another node type as well if we were to implement that?

I guess `BitcoindRpcClient` does implement `NodeApi` though so :man_shrugging:.

**Review by Christewart — APPROVED** [review]
Basically what I ended up leaving is nits, feel free to ignore if you want to get it merged

File: node/src/main/scala/org/bitcoins/node/NodeType.scala (hunk 2 above)
Comment (benthecarman) [reply node/src/main/scala/org/bitcoins/node/NodeType.scala:26]: I agree this is kinda bastardizing the `NodeType` but probably a better solution can be done in the future.

**Pull request merged**